File: Top 100 Codes/05-Operations on Strings/14-Capitalize the first and last character of each word of a string/Capitalizethefirstandlastcharacterofeachwordofastring.py
def capitalize_first_last(s):
    words = s.split()
    capitalized_words = []
    
    for word in words:
        if len(word) == 1:
            capitalized_words.append(word.upper())
        else:
            capitalized_word = word[0].upper() + word[1:-1] + word[-1].upper()
            capitalized_words.append(capitalized_word)
    
    return ' '.join(capitalized_words)

# Input from user
input_string = input("Enter a string: ")
result = capitalize_first_last(input_string)
print("Modified string:", result)

File: Top 100 Codes/05-Operations on Strings/14-Capitalize the first and last character of each word of a string/Capitalizethefirstandlastcharacterofeachwordofastring.cpp
#include<iostream>
#include<cctype>
#include<string>

using namespace std;

void capitalizeFirstLast(string &str){
    int n = str.length();

    for(int i = 0; i < n; i++){
        if(i == 0 || str[i -1] == ' '){
            str[i] = toupper(str[i]);
        }
        if(i < n-1 && str[i + 1] == ' '){
            str[i] = toupper(str[i]);
        }

        if(i = n-1){
            str[i] = toupper(str[i]);
        }
    }
}

int main(){
    string str;
    cout<<"Enter a String: ";
    getline(cin,str);

    capitalizeFirstLast(str);
    cout<<"Modified String: "<<str<<endl;

    return 0;
}